.circleci/config.yml

```diff
@@ -9,40 +9,39 @@ jobs:
             - checkout
             - restore_cache:
                   keys:
-                      - v2-dependencies-{{ checksum "package.json" }}-{{ checksum "mui/package.json" }}-{{ checksum "rn-vector/package.json"}}
+                      - v2-dependencies-{{ checksum "package.json" }}-{{ checksum "packages/mui/package.json" }}-{{ checksum "packages/rn-vector/package.json"}}
             - run:
                   name: Install Dependencies
-                  command: |
-                      yarn install
-                      cd rn-vector && yarn install && cd ..
-                      cd mui && yarn install
+                  command: yarn install --immutable
             - save_cache:
                   paths:
                       - node_modules
-                  key: v2-dependencies-{{ checksum "package.json" }}-{{ checksum "mui/package.json" }}-{{ checksum "rn-vector/package.json"}}
+                  key: v2-dependencies-{{ checksum "package.json" }}-{{ checksum "packages/mui/package.json" }}-{{ checksum "packages/rn-vector/package.json"}}
             - run:
-                  name: Build Font and SVG Icons
-                  command: yarn build
+                  name: Prettier Check
+                  command: yarn prettier:check
+            - run:
+                  name: Lint
+                  command: yarn lint
             - run:
-                  name: Build RN Vector Icons
-                  command: cd rn-vector && yarn build
+                  name: Build Font, SVG Icons, RN Vector Icons and MUI
+                  command: yarn build
             - run:
-                  name: Build MUI
-                  command: cd mui && yarn build
+                  name: Build React Demo
+                  command: |
+                      cd demos/react-demo
+                      yarn build-app
             - run:
-                  name: Verify SVG Build Artifacts
+                  name: Verify SVG Build and MUI Artifacts
```

Comment:
We should probably build the packages before building the demo (the demo depends on them and the demo build is also a slower step)